pkg/apis/core/types_shootsnapshot.go

```diff
@@ -0,0 +1,60 @@
+package core
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShootSnapshot contains a snapshot of the Shoot's state required to migrate the Shoot's control plane to a new Seed.
+type ShootSnapshot struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the ShootSnapshot.
+	Spec ShootSnapshotSpec
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ShootSnapshotList is a list of ShootSnapshot objects.
+type ShootSnapshotList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of ShootSnapshots.
+	Items []ShootSnapshot
+}
+
+// ShootSnapshotSpec is the specification of the ShootSnapshot.
+type ShootSnapshotSpec struct {
+	// Gardenlet holds the state of resources generated, deployed and reconciled by the gardenlet
+	Gardenlet GardenletResources
```

Comment:
Can we call that `GardenerResources` to not inject implementation details in the API?